src/transformers/models/auto/tokenization_auto.py

```diff
@@ -161,6 +161,7 @@
             ("ernie", ("BertTokenizer", "BertTokenizerFast" if is_tokenizers_available() else None)),
             ("ernie_m", ("ErnieMTokenizer" if is_sentencepiece_available() else None, None)),
             ("esm", ("EsmTokenizer", None)),
+            ("falcon", (None, "PreTrainedTokenizerFast" if is_tokenizers_available() else None)),
```

Comment:
I am confused. The config 

https://huggingface.co/tiiuae/falcon-rw-1b/blob/main/tokenizer_config.json

says it use `GPT2Tokenizer`? Why we use `PreTrainedTokenizerFast` here?

I must miss something in the discussion.

Reply:
(I am surprised however put `PreTrainedTokenizerFast` here solved the issue we have, but if the original `Falcon` use `GPT2Tokenizer`, that's the value to be put here).

Reply:
yeah this was based on another model that used the pretrainedTokenizerFast. If GPT2TokenizerFast and GPT2Tokenizer work let's add them

Reply:
Hi @ydshieh @arthurzucker, the `falcon-rw` models are more like 'prototype' models and are quite different from the real models. The real models use `PreTrainedTokenizerFast`: https://huggingface.co/tiiuae/falcon-40b/blob/main/tokenizer_config.json

Reply:
But at the end, the real models still use a concrete tokenizer class. Do you know which class it is? Then you can put it into the auto mapping.

Reply:
(or is the type is actually `PreTrainedTokenizerFast` when we load it from the checkpoint ...? If so, this is going to blow my mind)

Reply:
Oh ............that's so strange. OK, then good for me this PR. 

But this is not something good at all and we need to see why this happens and fix it.

Can't really have such non-information (`PreTrainedTokenizerFast` in auto mapping nor in tokenizer_config.json): it brings too much confusion and doesn't provide explicit information.

Reply:
I think we can if we explain that there is pretty much no extra layer of python behind this class, but I am biased because it doesn't seem bad for me. It's actually good if we have a single class wrapping everyone 😈

Reply:
But if this holds, it means no one can answer simple question easily like which tokenizer a model uses, say `persimmon`.
We have auto tokenizer thtat abstract this, which is very good, but the information should exist somewhere IMO ..?

I am still surprised how a class like `PreTrainedTokenizerFast` could be used directly, but now it looks like `falcon` (the real one) doesn't use any tokenizer defined in `transformers` (i.e. the concrete class). 

Of course, if this is really the case (i.e. it doesn't belong to any tokenizer class), it makes sense. But it's a bit strange why it is the case.

Reply:
Anyway , feel free to merge as long as CI 💚 .

Reply:
BTW, @Rocketknight1 You are the author of the port no? So I guess you know the history 😄 . When you port the tokenizer for falcon at the beginning, you are not using any concrete tokenizer class as a reference?


<img width="359" alt="Screenshot 2023-11-08 182816" src="https://github.com/huggingface/transformers/assets/2521628/60a56428-4cee-4bf1-b2f6-1ae9508ea122">

Reply:
I just copied all of that from the falcon custom code! You can see it at the [last commit before I updated everything](https://huggingface.co/tiiuae/falcon-40b/tree/c47b371b31a68349c233104050ac76680b8485db).

Reply:
OK, I think they might do sth strange ... Thanks for the update.